zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreterProcess.java

```diff
@@ -261,6 +261,12 @@ public int getNumIdleClient() {
     }
   }
 
+  public void setMaxPoolSize(int size) {
+    if (clientPool != null) {
+      //Size + 2 for progress poller , cancel operation
+      clientPool.setMaxTotal(size + 2);
```

Comment:
@Leemoonsoo The pool size of `GenearicObjectPool` is by default 8. Though the remotescheduler may depend on `maxPoolSize` for its queue management, it will not be able to schedule as many jobs since the pool size is limited.

For eg, if maxPoolSize is 10, the remoteScheduler can make 8 successful remote interpret call, remaining 2(which should have actually been scheduled on the remote interpreter) is blocked waiting on thread pool. Also if there is some cancel request, it will again be blocked waiting on thread pool. Same applies for JobstatusPoller, JobProgressPoller.

This is my understanding. Please correct me if I am being wrong.